command/healthcheck/pki_allow_if_modified_since.go

```diff
@@ -0,0 +1,103 @@
+package healthcheck
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-secure-stdlib/parseutil"
+)
+
+type AllowIfModifiedSince struct {
+	Enabled            bool
+	UnsupportedVersion bool
+
+	TuneData map[string]interface{}
+}
+
+func NewAllowIfModifiedSinceCheck() Check {
+	return &AllowIfModifiedSince{}
+}
+
+func (h *AllowIfModifiedSince) Name() string {
+	return "allow_if_modified_since"
+}
+
+func (h *AllowIfModifiedSince) IsEnabled() bool {
+	return h.Enabled
+}
+
+func (h *AllowIfModifiedSince) DefaultConfig() map[string]interface{} {
+	return map[string]interface{}{}
+}
+
+func (h *AllowIfModifiedSince) LoadConfig(config map[string]interface{}) error {
+	var err error
+
+	h.Enabled, err = parseutil.ParseBool(config["enabled"])
+	if err != nil {
+		return fmt.Errorf("error parsing %v.enabled: %w", h.Name(), err)
+	}
+
+	return nil
+}
+
+func (h *AllowIfModifiedSince) FetchResources(e *Executor) error {
+	exit, _, data, err := fetchMountTune(e, func() {
+		h.UnsupportedVersion = true
+	})
+	if exit {
+		return err
+	}
+
+	h.TuneData = data
+
+	return nil
+}
+
+func (h *AllowIfModifiedSince) Evaluate(e *Executor) (results []*Result, err error) {
+	if h.UnsupportedVersion {
+		ret := Result{
+			Status:   ResultInvalidVersion,
+			Endpoint: "/sys/mounts/{{mount}}/tune",
+			Message:  "This health check requires Vault 1.9+ but an earlier version of Vault Server was contacted, preventing this health check from running.",
+		}
+		return []*Result{&ret}, nil
+	}
+
+	req, err := stringList(h.TuneData["passthrough_request_headers"])
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse value from server for passthrough_request_headers: %w", err)
+	}
+
+	resp, err := stringList(h.TuneData["allowed_response_headers"])
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse value from server for allowed_response_headers: %w", err)
+	}
+
+	var foundIMS bool = false
+	for _, param := range req {
+		if strings.EqualFold(param, "If-Modified-Since") {
+			foundIMS = true
+			break
+		}
+	}
+
+	var foundLM bool = false
+	for _, param := range resp {
+		if strings.EqualFold(param, "Last-Modified") {
+			foundLM = true
+			break
+		}
+	}
+
+	if !foundIMS || !foundLM {
```

Comment:
I imagine there isn't a way for us to detect they are running a Vault Server version that would honor these headers?

Reply:
We could maybe infer it by querying health check, but I think that might require some more machinery and depend on execution order of these health checks... We don't surface or aggregate the unsupported version errors...

Reply:
Per OOB discussion, we can proactively query `http://127.0.0.1:8200/v1/sys/seal-status` in the future for the server version if we desire.